src/include/btmem.h

```diff
@@ -1131,8 +1132,33 @@ struct __wt_update {
      */
     volatile uint8_t prepare_state; /* prepare state */
 
+/*
+ * WT_UPDATE_DS indicates that an update is intended to be written to the data store during a round
+ * of reconciliation. If the flag is set on an update it indicates that subsequent rounds of
+ * reconciliation can write that update even if they can't see it. This is done to prevent an older
+ * update in the update chain overwriting the newer on-page update. Ideally this flag would only
+ * be set if we knew that the reconciliation would be successful, however we cannot know that.  As
+ * such the reconciliation can still fail after setting the flag, this is okay. However be extremely
+ * careful when adding logic to the code that uses this flag. Generally the reconciliation will only
+ * fail when a mixed-mode or out-of-order timestamp update is encountered by eviction and a
+ * checkpoint is running concurrently.
+ *
+ * It is only flagged on the on-page update if ALL older updates were successfully wrote to the
+ * history store in the history store insert path. It may also be flagged on the tombstone
+ * preceding the on-page update if one exists in the same code path.
+ *
+ * There are two other scenarios where we could have marked the on page update as required to be
+ * written to the datastore, they are being listed here for completeness:
+ *  - When we select a tombstone to write to the data store, in this case no entries are
```

Comment:
This refers to my other comments. Can we save the tombstone in the new tombstone field?

Reply:
Ah yes that would be better IMO.

Reply:
After a discussion we have chosen not to go with this due to there being two different structures with somewhat different meanings, i.e. WT_UPDATE_SELECT and WT_SAVE_UPD.